examples/ios/objc/Draw/AppDelegate.m

```diff
@@ -0,0 +1,84 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import "AppDelegate.h"
+#import <Realm/Realm.h>
+#import "DrawView.h"
+#import "Constants.h"
+
+@implementation AppDelegate
+
+- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
+{
+    NSString *ipAddress = kIPAddress;
+
+    application.applicationSupportsShakeToEdit = YES;
+    
+    // Setup Global Error Handler
+    RLMSyncErrorReportingBlock globalErrorHandler = ^(NSError *error, RLMSyncSession *session) {
+        NSLog(@"A global error has occurred! %@", error);
+    };
+    
+    [RLMSyncManager sharedManager].errorHandler = globalErrorHandler;
+    
+    if ([RLMSyncUser allUsers].count > 0) {
+        NSURL *syncURL = [NSURL URLWithString:[NSString stringWithFormat:@"realm://%@:9080/~/Draw", ipAddress]];
+        RLMSyncConfiguration *syncConfig = [[RLMSyncConfiguration alloc] initWithUser:[RLMSyncUser currentUser] realmURL:syncURL];
+        RLMRealmConfiguration *defaultConfig = [RLMRealmConfiguration defaultConfiguration];
+        defaultConfig.syncConfiguration = syncConfig;
+        [RLMRealmConfiguration setDefaultConfiguration:defaultConfig];
+    }
+    else {
+        // The base server path
+        // Set to connect to local or online host
+        NSURL *authURL = [NSURL URLWithString:[NSString stringWithFormat:@"http://%@:9080", ipAddress]];
+        
+        // Creating a debug credential since this demo is just using the generated access token
+        // produced when running the Realm Object Server via the `start-object-server.command`
+        RLMSyncCredentials *credential = [RLMSyncCredentials credentialsWithUsername:@"demo@realm.io"
+                                                                         password:@"password"
+                                                                          register:NO];
+        
+        // Log the user in (async, the Realm will start syncing once the user is logged in automatically)
+        [RLMSyncUser logInWithCredentials:credential
+                            authServerURL:authURL
+                             onCompletion:^(RLMSyncUser *user, NSError *error) {
+           if (error) {
+               NSLog(@"A login error has occurred! %@", error);
+           }
+           else { // Logged in setup the default Realm
+               // The Realm virtual path on the server.
+               // The `~` represents the Realm user ID. Since the user ID is not known until you
+               // log in, the ~ is used as short-hand to represent this.
+               NSURL *syncURL = [NSURL URLWithString:[NSString stringWithFormat:@"realm://%@:9080/~/Draw", ipAddress]];
```

Comment:
These 5 lines are identical to the `if ([RLMSyncUser allUsers].count > 0) {` case above. Extract into a method?